### PR TITLE
K8s: Add apimachinery and apiserver packages

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

This adds `pkg/apimachinery` and `pkg/apiserver` packages with independent `go.mod` files. These new packages are also added to the go workspace introduced in https://github.com/grafana/grafana/pull/83191.

Related PR: https://github.com/grafana/grafana-enterprise/pull/6308

**Why do we need this feature?**

This is to help prepare for making apps separate packages.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: go.work [hunk 1]
```diff
@@ -0,0 +1,7 @@
+go 1.22.0
```
Comment (itsmylife) [inline go.work:1]: go.mod has version 1.21.0 Do we need them both in same version?

Comment (toddtreece) [reply go.work]: updated 👍

**Review by charandas — APPROVED** [review]
Nice work!

**Pull request merged**